debezium-core/src/main/java/io/debezium/pipeline/source/snapshot/incremental/AbstractIncrementalSnapshotChangeEventSource.java

```diff
@@ -100,6 +100,25 @@ public void closeWindow(P partition, String id, OffsetContext offsetContext) thr
         readChunk(partition);
     }
 
+    @Override
+    public void pauseSnapshot(OffsetContext offsetContext) throws InterruptedException {
+        context = (IncrementalSnapshotContext<T>) offsetContext.getIncrementalSnapshotContext();
+        if (context.snapshotRunning() && !context.isSnapshotPaused()) {
+            context.pauseSnapshot();
+        }
+    }
+
+    @Override
+    public void resumeSnapshot(P partition, OffsetContext offsetContext) throws InterruptedException {
+        context = (IncrementalSnapshotContext<T>) offsetContext.getIncrementalSnapshotContext();
+        if (context.snapshotRunning() && context.isSnapshotPaused()) {
+            context.resumeSnapshot();
+            window.clear();
```

Comment:
This code is quite similar to `reReadChunk()`. Cannot it be re-used? At the same time, should all those operations be called? IIRC the pause is done after the current chunk is processed, is not it? If yes then this cleanup should be unnecessary.

Reply:
I had some issue when using `reReadChunk()`. I'll investigate if the cleanup can be removed - this would be the best way, reuse `reReadChunk()` second one.

Reply:
yes, the cleanup is not needed